### PR TITLE
Create `Organization` Entities

### PR DESCRIPTION
**Pull request by rashidovasadbek**
Closes: #7

### CODE REVIEW

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/Organization.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+﻿using Feedback.Analyzer.Domain.Common.Entities;
+
+namespace Feedback.Analyzer.Domain.Entities;
+
+/// <summary>
+/// Represents an organization within the system.
+/// </summary>
+/// <remarks>
+/// Inherits from SoftDeletedEntity to support soft deletion functionality.
+/// </remarks>
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/Organization.cs:10]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/Organization.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+{
+    /// <summary>
+    /// The unique identifier of the client associated with this organization.
+    /// </summary>
+    public Guid ClientId { get; set; }
+    
+    /// <summary>
+    /// Gets or sets the  name of the Organization.
+    /// </summary>
+    public string? OrganizationName { get; set; }
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/Organization.cs:21]: Change to "Name" and make it not nullable, recreate migration

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/Organization.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    
+    /// <summary>
+    ///  Gets or sets the  description of the Organization.
+    /// </summary>
+    public string? Description { get; set; }
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/Organization.cs:26]: Make it not nullable, recreate migration

**Review by botirjonabdumannonovv — APPROVED** [review]

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/Interfaces/IOrganizationRepository.cs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    ValueTask<Organization?> GetByIdAsync(Guid organizationId, bool asNoTracking = false, CancellationToken cancellationToken = default);
+    
+    /// <summary>
+    /// Retrieves a list of Organizations based on a collection of their unique identifiers (IDs).
+    /// </summary>
+    /// <param name="ids">A collection of Organization IDs.</param>
+    /// <param name="asNoTracking">If true, disables change tracking for the returned entities, potentially improving performance.</param>
+    /// <param name="cancellationToken">A token to allow the operation to be cancelled.</param>
+    /// <returns>A list of Organizations matching the provided IDs.  If an ID doesn't correspond to an existing Organization, it is omitted in the result.</returns> 
+    ValueTask<IList<Organization>> GetByIdsAsync(IEnumerable<Guid> ids, bool asNoTracking = false, CancellationToken cancellationToken = default);
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/Interfaces/IOrganizationRepository.cs:36]: Add `QueryOptions`

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/Interfaces/IOrganizationRepository.cs [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    ValueTask<Organization?> DeleteAsync(Organization organization, bool saveChanges = true, CancellationToken cancellationToken = default);
+    
+    /// <summary>
+    /// Deletes an existing Organization record based on its unique identifier (ID).
+    /// </summary>
+    /// <param name="organizationId">The ID of the Organization to be deleted.</param>
+    /// <param name="saveChanges">If true, automatically saves changes to the underlying data store. If false, additional changes can be made before saving.</param>
+    /// <param name="cancellationToken">A token to allow the operation to be cancelled.</param>
+    /// <returns>The deleted Organization object if successful, otherwise null (e.g., if the Organization was not found).</returns>
+    ValueTask<Organization?> DeleteByIdAsync(Guid organizationId, bool saveChanges = true, CancellationToken cancellationToken = default);
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/Interfaces/IOrganizationRepository.cs:76]: Add `CommandOptions`

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/Interfaces/IOrganizationRepository.cs [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    ValueTask<IList<Organization>> GetByIdsAsync(IEnumerable<Guid> ids, bool asNoTracking = false, CancellationToken cancellationToken = default);
+    
+    /// <summary>
+    /// Creates a new Organization record.
+    /// </summary>
+    /// <param name="organization">The Organization object to be created.</param>
+    /// <param name="saveChanges">If true, automatically saves changes to the underlying data store. If false, additional changes can be made before saving.</param>
+    /// <param name="cancellationToken">A token to allow the operation to be cancelled.</param>
+    /// <returns>The newly created Organization object.</returns>
+    /// <exception cref="ArgumentException">Thrown if the provided 'organization' has a default (empty) ID or if an organization with the same ID already exists.</exception>
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/Interfaces/IOrganizationRepository.cs:45]: Remove exception

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/Interfaces/IOrganizationRepository.cs [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    
+    /// <summary>
+    /// Updates an existing Organization record.
+    /// </summary>
+    /// <param name="organization">The Organization object containing modified data.</param>
+    /// <param name="saveChanges">If true, automatically saves changes to the underlying data store. If false, additional changes can be made before saving.</param>
+    /// <param name="cancellationToken">A token to allow the operation to be cancelled.</param>
+    /// <returns>The updated Organization object.</returns> 
+    /// <exception cref="ArgumentNullException">Thrown if the provided 'organization' is null.</exception>
+    /// <exception cref="InvalidOperationException">Thrown if the Organization cannot be found for the provided ID.</exception>
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/Interfaces/IOrganizationRepository.cs:56]: Remove exception

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/Interfaces/IOrganizationRepository.cs [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    ValueTask<Organization> UpdateAsync(Organization organization, bool saveChanges = true, CancellationToken cancellationToken = default);
+    
+    /// <summary>
+    /// Deletes an existing Organization record.
+    /// </summary>
+    /// <param name="organization">The Organization object to be deleted.</param>
+    /// <param name="saveChanges">If true, automatically saves changes to the underlying data store. If false, additional changes can be made before saving.</param>
+    /// <param name="cancellationToken">A token to allow the operation to be cancelled.</param>
+    /// <returns>The deleted Organization object if successful, otherwise null (e.g., if the Organization was not found).</returns>
+    /// <exception cref="ArgumentNullException">Thrown if the provided 'organization' is null.</exception>
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/Interfaces/IOrganizationRepository.cs:66]: Remove exception

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/OrganizationRepository.cs [hunk 9]
```diff
@@ -0,0 +1,34 @@
+﻿using System.Linq.Expressions;
+using Feedback.Analyzer.Domain.Entities;
+using Feedback.Analyzer.Persistence.DataContexts;
+using Feedback.Analyzer.Persistence.Repositories.Interfaces;
+
+namespace Feedback.Analyzer.Persistence.Repositories;
+
+public class OrganizationRepository(AppDbContext dbContext)
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/OrganizationRepository.cs:8]: Add type docs comment

**Review by sulton-max — CHANGES_REQUESTED** [review]

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Commands/CommandOptions.cs [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+{
+    /// <summary>
+    /// Gets or sets a value indicating whether changes made by the command should be automatically saved to the underlying data store.
+    /// </summary>
+    public bool SaveChanges { get; set; }
+
+    /// <summary>
+    /// Provides a default set of CommandOptions, likely with 'SaveChanges' set to true.
+    /// </summary>
+    public static CommandOptions Default => new();
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Commands/CommandOptions.cs:16]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Queries/QueryOptions.cs [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+    /// <summary>
+    /// Gets or sets a value indicating whether change tracking should be disabled for query results. 
+    /// Disabling change tracking can potentially improve performance.
+    /// </summary>
+    public bool AsNoTracking { get; set; }
+    
+    /// <summary>
+    /// Provides a default set of QueryOptions, likely with 'AsNoTracking' set to false.
+    /// </summary>
+    public static QueryOptions Default => new ();
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Queries/QueryOptions.cs:17]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/OrganizationRepository.cs [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+using Feedback.Analyzer.Domain.Entities;
+using Feedback.Analyzer.Persistence.DataContexts;
+using Feedback.Analyzer.Persistence.Repositories.Interfaces;
+
+namespace Feedback.Analyzer.Persistence.Repositories;
+
+/// <summary>
+/// Provides data access functionality for Organization entities, 
+/// inheriting core repository logic from EntityRepositoryBase.
+/// Implements the IOrganizationRepository interface.
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/OrganizationRepository.cs:13]: Remove

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by sulton-max — APPROVED** [review]
Well done

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Migrations/20240226085026_Add_Organization Entity.cs [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+                oldType: "character varying(256)",
+                oldMaxLength: 256);
+
+            migrationBuilder.CreateTable(
+                name: "Organizations",
+                columns: table => new
+                {
+                    Id = table.Column<Guid>(type: "uuid", nullable: false),
+                    ClientId = table.Column<Guid>(type: "uuid", nullable: false),
+                    OrganizationName = table.Column<string>(type: "character varying(128)", maxLength: 128, nullable: false),
```
Comment (isxoqovxasanboy) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Migrations/20240226085026_Add_Organization Entity.cs:40]: OrganizationName change Name

**Review by sulton-max — APPROVED** [review]

**Review by botirjonabdumannonovv — APPROVED** [review]

**Review by isxoqovxasanboy — APPROVED** [review]

**Pull request merged**